controllers/clustertopology_controller_compute.go

```diff
@@ -60,7 +62,16 @@ type clusterTopologyState struct {
 	cluster               *clusterv1.Cluster
 	infrastructureCluster *unstructured.Unstructured
 	controlPlane          controlPlaneTopologyState
-	machineDeployments    []machineDeploymentTopologyState //nolint:structcheck
+	machineDeployments    []machineDeploymentTopologyState
```

Comment:
@fabriziopandini @sbueringer @killianmuldoon Does it make sense to convert the array to a map of topologyName <-> machineDeploymentTopologyState.
That would get rid of the extra method we have here, what say?

Reply:
I like it as it removes the need in my PR to generate the map :)
Note: this also affects @killianmuldoon 's PR

Reply:
Hence I tagged him on this, I will update my PR to change it to `map[string]machineDeploymentTopologyState`. And work with @killianmuldoon to make sure there are no breaking surprises.

Reply:
we are moving this change to #5045 (see https://github.com/kubernetes-sigs/cluster-api/pull/5045#pullrequestreview-727324599)
@srm09 thanks for raising this issue!

Reply:
Will rebase this PR once #5045 gets merged in.